File: Heat/Controllers/FuelsController.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.Entity
Imports System.Linq
Imports System.Net
Imports System.Web
Imports System.Web.Mvc
Imports Heat
Imports Heat.Models
Imports Heat.Repositories
Imports System.Security.Claims
Imports log4net
 
Namespace Controllers
    Public Class FuelsController
        Inherits System.Web.Mvc.Controller

        Public db As New HeatDBContext
         
        ' GET: Fuels
        '<Authorize(roles:="canViewFuels")> _
        <ClaimsAutorize(ClaimTypes.Name, "demo")> _
        Function Index() As ActionResult
            Return View(db.Fuels.ToList())
        End Function

        ' GET: Fuels/Details/5
        <ResourceAuthorize(ResourceOperations.ReadOwn, "FuelDetail")> _
        Function Details(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim fuel As Fuel = db.Fuels.Find(id)
            If IsNothing(fuel) Then
                Return HttpNotFound()
            End If
            Return View(fuel)
        End Function

        ' GET: Fuels/Create
        Function Create() As ActionResult
            Return View()
        End Function

        ' POST: Fuels/Create
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Create(<Bind(Include:="ID,Name")> ByVal fuel As Fuel) As ActionResult
            If ModelState.IsValid Then
                db.Fuels.Add(fuel)
                db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            Return View(fuel)
        End Function

        ' GET: Fuels/Edit/5
        Function Edit(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim fuel As Fuel = db.Fuels.Find(id)
            If IsNothing(fuel) Then
                Return HttpNotFound()
            End If
            Return View(fuel)
        End Function

        ' POST: Fuels/Edit/5
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Edit(<Bind(Include:="ID,Name")> ByVal fuel As Fuel) As ActionResult
            If ModelState.IsValid Then
                db.Entry(fuel).State = EntityState.Modified
                db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            Return View(fuel)
        End Function

        ' GET: Fuels/Delete/5
        Function Delete(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim fuel As Fuel = db.Fuels.Find(id)
            If IsNothing(fuel) Then
                Return HttpNotFound()
            End If
            Return View(fuel)
        End Function

        ' POST: Fuels/Delete/5
        <HttpPost()>
        <ActionName("Delete")>
        <ValidateAntiForgeryToken()>
        Function DeleteConfirmed(ByVal id As Integer) As ActionResult
            Dim fuel As Fuel = db.Fuels.Find(id)
            db.Fuels.Remove(fuel)
            db.SaveChanges()
            Return RedirectToAction("Index")
        End Function

        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If (disposing) Then
                db.Dispose()
            End If
            MyBase.Dispose(disposing)
        End Sub
    End Class
End Namespace

File: Heat.ConvertedToC#/Controllers/FuelsController.cs
using System.Linq;
using System.Web.Mvc;
using System.Data.Entity;
using System.Net;
using Heat.Models;
using Heat.Repositories;
using System.Security.Claims;
using System.IdentityModel.Services;

namespace Heat.Controllers
{
    public class FuelsController : System.Web.Mvc.Controller
	{


		public HeatDBContext db = new HeatDBContext();
		// GET: Fuels
		//<Authorize(roles:="canViewFuels")> _
		[ClaimsAutorize(ClaimTypes.Name, "demo")]
		public ActionResult Index()
		{
			return View(db.Fuels.ToList());
		}

		// GET: Fuels/Details/5
		[ResourceAuthorize(ResourceOperations.ReadOwn, "FuelDetail")]
		public ActionResult Details(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			Fuel fuel = db.Fuels.Find(id);
			if ((fuel == null)) {
				return HttpNotFound();
			}
			return View(fuel);
		}

		// GET: Fuels/Create
		[ClaimsPrincipalPermission(System.Security.Permissions.SecurityAction.Demand, Operation = "Create", Resource = "Fuel")]
		public ActionResult Create()
		{
			return View();
		}

		// POST: Fuels/Create
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Create(		[Bind(Include = "ID,Name")]
Fuel fuel)
		{
			if (ModelState.IsValid) {
				db.Fuels.Add(fuel);
				db.SaveChanges();
				return RedirectToAction("Index");
			}
			return View(fuel);
		}

		// GET: Fuels/Edit/5
		public ActionResult Edit(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			Fuel fuel = db.Fuels.Find(id);
			if ((fuel == null)) {
				return HttpNotFound();
			}
			return View(fuel);
		}

		// POST: Fuels/Edit/5
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Edit(		[Bind(Include = "ID,Name")]
Fuel fuel)
		{
			if (ModelState.IsValid) {
				db.Entry(fuel).State = EntityState.Modified;
				db.SaveChanges();
				return RedirectToAction("Index");
			}
			return View(fuel);
		}

		// GET: Fuels/Delete/5
		public ActionResult Delete(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			Fuel fuel = db.Fuels.Find(id);
			if ((fuel == null)) {
				return HttpNotFound();
			}
			return View(fuel);
		}

		// POST: Fuels/Delete/5
		[HttpPost()]
		[ActionName("Delete")]
		[ValidateAntiForgeryToken()]
		public ActionResult DeleteConfirmed(int id)
		{
			Fuel fuel = db.Fuels.Find(id);
			db.Fuels.Remove(fuel);
			db.SaveChanges();
			return RedirectToAction("Index");
		}

		[HttpGet()]
		public FileResult PrintToPDF()
		{
			return File("c:\\temp\\prova.pdf", "application/pdf");
		}
		protected override void Dispose(bool disposing)
		{
			if ((disposing)) {
				db.Dispose();
			}
			base.Dispose(disposing);
		}
	}
}
